common/db/migrations/088_rights_management.rb

```diff
@@ -0,0 +1,712 @@
+        :notes => ASUtils.to_json({
+                                    'jsonmodel_type' => 'note_agent_rights_statement',
+                                    'content' => [note_content],
+                                    'persistent_id' => SecureRandom.hex
+                                  }),
+        :last_modified_by => row[:last_modified_by],
+        :create_time => row[:create_time],
+        :system_mtime => row[:system_mtime],
+        :user_mtime => row[:user_mtime]
+      )
+    end
+
+    # link any agent rights statement external documents to the agent instead
+    self[:external_document]
+      .join(:rights_statement, :rights_statement__id => :external_document__rights_statement_id)
+      .filter(Sequel.~(Sequel.qualify(:rights_statement, agent_fk) => nil))
+      .select(Sequel.as(:external_document__id, :external_document_id),
+              Sequel.as(Sequel.qualify(:rights_statement, agent_fk), agent_fk))
+      .each do |row|
+
+      self[:external_document]
+        .filter(:id => row[:external_document_id])
+        .update(:rights_statement_id => nil,
+                :identifier_type_id => nil,
+                agent_fk => row[agent_fk])
+    end
+  end
+end
+
+
+def drop_old_rights_statement_columns
+  alter_table(:rights_statement) do
+    # drop fks
+    drop_foreign_key [:ip_status_id] #, :name => :rights_statement_ibfk_2
+
+    # drop columns
+    drop_column(:active)
+    drop_column(:ip_status_id)
+    drop_column(:restriction_start_date)
+    drop_column(:restriction_end_date)
+    drop_column(:materials)
+    drop_column(:ip_expiration_date)
+    drop_column(:type_note)
+    drop_column(:permissions)
+    drop_column(:restrictions)
+    drop_column(:granted_note)
+    drop_column(:license_identifier_terms)
+  end
+end
+
+
+def drop_agent_rights_statements
+  self[:rights_statement]
+    .filter(Sequel.|(
+      Sequel.~(:agent_person_id => nil),
+      Sequel.~(:agent_family_id => nil),
+      Sequel.~(:agent_corporate_entity_id => nil),
+      Sequel.~(:agent_software_id => nil)
+    ))
+    .delete
+
+  alter_table(:rights_statement) do
+    drop_foreign_key [:agent_person_id]
+    drop_foreign_key [:agent_family_id]
+    drop_foreign_key [:agent_corporate_entity_id]
+    drop_foreign_key [:agent_software_id]
+
+    drop_column(:agent_person_id)
+    drop_column(:agent_family_id)
+    drop_column(:agent_corporate_entity_id)
+    drop_column(:agent_software_id)
+  end
+end
+
+
+def coerce_date(date)
+  if date =~ /[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]/
+    date # Date.strptime(date, '%Y-%m-%d')
+  elsif date =~ /[0-9][0-9][0-9][0-9]-[0-9][0-9]/
+    "#{date}-01" # Date.strptime("#{date}-01", '%Y-%m-%d')
+  elsif date =~ /[0-9][0-9][0-9][0-9]/
+    "#{date}-01-01" # Date.strptime("#{date}-01-01", '%Y-%m-%d')
+  else
+    raise "Not a date: #{date}"
+  end
+end
+
+
+def coerce_timestamp(timestamp)
+  timestamp.strftime('%Y-%m-%d')
+end
+
+Sequel.migration do
+
+  up do
+    self.transaction do
```

Comment:
Now with extra pizazz! We `backup_rights_statement_table` before we touch any of the data.